x/records/module_ibc.go

```diff
@@ -168,44 +167,33 @@ func (im IBCModule) OnAcknowledgementPacket(
 		return sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-20 transfer packet acknowledgement: %v", err)
 	}
-	var data ibctransfertypes.FungibleTokenPacketData
-	if err := ibctransfertypes.ModuleCdc.UnmarshalJSON(packet.GetData(), &data); err != nil {
-		im.keeper.Logger(ctx).Error(fmt.Sprintf("Error unmarshalling packet  %v", err.Error()))
-		return sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-20 transfer packet data: %s", err.Error())
-	}
 
 	// Custom ack logic only applies to ibc transfers initiated from the `stakeibc` module account
 	// NOTE: if the `stakeibc` module account IBC transfers tokens for some other reason in the future,
 	// this will need to be updated
-	if data.Sender == im.keeper.AccountKeeper.GetModuleAddress(stakeibctypes.ModuleName).String() {
-		switch resp := ack.Response.(type) {
-		case *channeltypes.Acknowledgement_Result:
-			im.keeper.Logger(ctx).Info(fmt.Sprintf("[IBC-TRANSFER] Acknowledgement_Result {%s}", string(resp.Result)))
-			// UPDATE RECORD
-			// match record based on amount
-			amount, err := strconv.ParseInt(data.Amount, 10, 64)
-			if err != nil {
-				return sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "Error parsing int %d", amount)
-			}
-			record, found := im.keeper.GetTransferDepositRecordByAmount(ctx, amount)
-			if !found {
-				return sdkerrors.Wrapf(sdkerrors.ErrNotFound, "No deposit record found for amount: %d", amount)
-			}
-			// update the record
-			record.Status = types.DepositRecord_STAKE
-			im.keeper.SetDepositRecord(ctx, *record)
-			im.keeper.Logger(ctx).Info(fmt.Sprintf("[IBC-TRANSFER] Deposit record updated to STAKE: {%v}", record))
-			im.keeper.Logger(ctx).Info(fmt.Sprintf("[IBC-TRANSFER] success to %s", record.HostZoneId))
-		case *channeltypes.Acknowledgement_Error:
-			im.keeper.Logger(ctx).Error(fmt.Sprintf("[IBC-TRANSFER] Acknowledgement_Error {%s}", resp.Error))
+	switch resp := ack.Response.(type) {
+	case *channeltypes.Acknowledgement_Result:
+		im.keeper.Logger(ctx).Info(fmt.Sprintf("\t [IBC-TRANSFER] Acknowledgement_Result {%s}", string(resp.Result)))
+		// callback
+		err := im.keeper.ICACallbacksKeeper.CallRegisteredICACallback(ctx, packet, &ack)
+		if err != nil {
+			errMsg := fmt.Sprintf("Unable to call registered callback from records OnAcknowledgePacket | Sequence %d, from %s %s, to %s %s | Error %s",
+				packet.Sequence, packet.SourceChannel, packet.SourcePort, packet.DestinationChannel, packet.DestinationPort, err.Error())
+			im.keeper.Logger(ctx).Error(errMsg)
+			return sdkerrors.Wrapf(icacallbacktypes.ErrCallbackFailed, errMsg)
 		}
+	case *channeltypes.Acknowledgement_Error:
+		im.keeper.Logger(ctx).Error(fmt.Sprintf("\t [IBC-TRANSFER] Acknowledgement_Error {%s}", resp.Error))
+	default:
+		im.keeper.Logger(ctx).Error(fmt.Sprintf("\t [IBC-TRANSFER] Unrecognized ack for packet {%v}", packet))
 	}
 
 	return im.app.OnAcknowledgementPacket(ctx, packet, acknowledgement, relayer)
```

Comment:
If the transfer ack was handled at the top here, do we still want to pass this ack down to stakeibc? 
Seems like if `im.keeper.ICACallbacksKeeper.CallRegisteredICACallback` succeeds, we should just return?